### DISCUSSION TITLE
Can we add the gem `web-console` to the project?

### DISCUSSION THREAD

**Discussion by KarlHeitmann** [joemasilotti/railsdevs.com#655] [Ideas]
Hi!

What do you think about adding the gem [web-console](https://github.com/rails/web-console) as a dev dependency for the project?

I like the policy to keep  the Gemfile as slim as possible, but I think this gem is a must-to  in every project. I use it not only to check what went wrong in case an unexpected problem rises while working.When I am exploring a project, I sometimes break the code on purpose to use `web-console` as a debugger.

When I don't know what is going on, or want to inspect a variable, instead putting a `binding.pry` or a `puts` statement, I put a division by zero line: `1/0`, and when the interpreter arrives to the division by zero line, then I use the console on the browser to experiment with the variables on the scope where the division by zero occurred. Very useful to tinker with variables, experiment and debug.

I've already used the Gem in this project while working on #627 but I kept my Gemfile away of the commits.

**Comment by joemasilotti** [comment]
I've never used web-console before. What benefit does it have over `binding.pry`?

**Reply by KarlHeitmann** [reply]
I use it in a hacky way as a substitute -sometimes- for `binding.pry`, but it's intended to use when an error occurs in the browser: it shows you a console on the browser with the context right before the error occured. Take a look at this example, I did an error on purpose here `DevelopersController#index`

![image](https://user-images.githubusercontent.com/3003032/190838056-761f4c89-7299-44fe-b497-483d1ad59721.png)

When I visit `/developers` page in the browser, I will get an error.

![example_web_consle](https://user-images.githubusercontent.com/3003032/190838195-93721f0e-585d-4639-b13b-985cd27a8d24.png)

Notice the bottom of the page: you have a console where you can type the variables in the context where the error occured. That console is the same console you get when you put a `binding.pry` sentence, or when you run `rails c`

`web-console` gem is part of the default rails stack, I don't remember now, but I think since rails 5.x, when you create a rails app using `rails new` you will get that gem automatically added to your `Gemfile` in the `:development` section of it.

**Reply by joemasilotti** [reply]
Got it, thanks for the explanation! If `web-console` is part of the default Rails stack then let's go ahead and add it to RailsDevs.
